zlog.c

```diff
@@ -135,9 +137,14 @@ static void* zlog_buffer_flush_thread()
 }
 
 void zlog_init_flush_thread(void)
+{
+    zlog_init_flush_thread_with_interval(ZLOG_FLUSH_INTERVAL_SEC);
+}
+
+void zlog_init_flush_thread_with_interval(int interval_sec)
 {
     pthread_t thr;
-    pthread_create(&thr, NULL, zlog_buffer_flush_thread, NULL);
+    pthread_create(&thr, NULL, zlog_buffer_flush_thread, (void*)(intptr_t)interval_sec);
```

Comment:
`malloc` from heap for storing the `arg` passed to `pthread_create` will be safer.